zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -63,13 +63,14 @@ public boolean isAuthorized(final AuthorizationRequest request) {
       return true;
     }
 
-    Set<String> authorizedResourceIdentifiers = new HashSet<>();
-
+    final Stream<String> authorizedResourceIdentifiers;
     final var userKey = getUserKey(request);
     if (userKey.isPresent()) {
       authorizedResourceIdentifiers =
           getUserAuthorizedResourceIdentifiers(
               userKey.get(), request.getResourceType(), request.getPermissionType());
+    } else {
+      authorizedResourceIdentifiers = Stream.empty();
     }
```

Comment:
```suggestion
    Stream<String> authorizedResourceIdentifiers = Stream.empty();
    final var userKey = getUserKey(request);
    if (userKey.isPresent()) {
      authorizedResourceIdentifiers =
          getUserAuthorizedResourceIdentifiers(
              userKey.get(), request.getResourceType(), request.getPermissionType());
    }
```

🔧 We can get rid of the else block here

Reply:
That was actually a conscious choice because the var can be final if it is immediately initialized in both the if and the else branch.

Reply:
I figured it might've been. I have not strong preference for one over the other tbh